propagators/ot/ot_propagator.go

```diff
@@ -93,18 +94,42 @@ func (o OT) Extract(ctx context.Context, carrier propagation.TextMapCarrier) con
 	if err != nil || !sc.IsValid() {
 		return ctx
 	}
-	// TODO: implement extracting baggage
-	//
-	// this currently is not achievable without an implementation of `keys`
-	// on the carrier, see:
-	// https://github.com/open-telemetry/opentelemetry-go/issues/1493
+
+	bags, err := extractBags(carrier)
+	if err != nil {
+		return trace.ContextWithRemoteSpanContext(ctx, sc)
+	}
+	ctx = baggage.ContextWithBaggage(ctx, bags)
 	return trace.ContextWithRemoteSpanContext(ctx, sc)
 }
 
 func (o OT) Fields() []string {
 	return []string{traceIDHeader, spanIDHeader, sampledHeader}
 }
 
+// extractBags reconstructs the baggage information from opentracing
+func extractBags(carrier propagation.TextMapCarrier) (baggage.Baggage, error) {
+	emptyBags, _ := baggage.New()
+	var members []baggage.Member
+	for _, key := range carrier.Keys() {
+		lowerKey := strings.ToLower(key)
+		if !strings.HasPrefix(lowerKey, baggageHeaderPrefix) {
+			continue
+		}
+		strippedKey := strings.TrimPrefix(lowerKey, baggageHeaderPrefix)
+		member, err := baggage.NewMember(strippedKey, carrier.Get(key))
+		if err != nil {
+			return emptyBags, err
```

Comment:
Should this accumulate into a `multierr.Error` and just drop the bad entry rather than potentially losing all baggage members?

Reply:
I was thinking the same thing. Will update with that change.